src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -201,272 +177,248 @@ public ApartmentValueFunction() {
 
 	/**
 	 * Set the function which will be used to calculate the subjective value of the
-	 * floor area
-	 * 
+	 * floor area.
+	 *
 	 * @param floorAreaValueFunction an object of type {@link PartialValueFunction}
-	 * @throws IllegalArgumentException
 	 */
-	public void setFloorAreaValueFunction(PartialValueFunction<Double> floorAreaValueFunction)
-			throws IllegalArgumentException {
+	public void setFloorAreaValueFunction(PartialValueFunction<Double> floorAreaValueFunction) {
+		checkNotNull(floorAreaValueFunction, "The floor area preferencies cannot be null");
 		this.floorAreaValueFunction = floorAreaValueFunction;
```

Comment:
`this.floorAreaValueFunction = checkNotNull(floorAreaValueFunction);` suffit (OPT)

Reply:
Fixed in commit 9e2baa6c4f3562050ce514a25a48fe2bf139184e